src/python_testing/TC_CNET_4_10.py

```diff
@@ -0,0 +1,422 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+import test_plan_support
+from chip.clusters.Types import NullValue
+from chip.testing.decorators import has_feature, run_if_endpoint_matches
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main
+from mobly import asserts
+
+# Thread TLV Type mapping for operational dataset parsing
+THREAD_TLV_TYPE_MAP = {
+    0x00: "Channel",
+    0x01: "PanId",
+    0x02: "ExtendedPanId",
+    0x03: "NetworkName",
+    0x04: "PSKc",
+    0x05: "NetworkKey",
+    0x07: "MeshLocalPrefix",
+    0x0c: "SecurityPolicy",
+    0x0e: "ActiveTimestamp",
+    0x35: "ChannelMask",
+    0x4a: "WakeupChannel"
+}
+
+
+def parse_openthread_dataset_stream(dataset_hex: str) -> dict[str, str] | None:
```

Comment:
This is cool. Thanks. Are there other tests where this would be useful? This might actually be interesting to extract out into its own file. Would you mind doing that as a follow up? Seems like it could be a good utility function generally.